pkg/sdk/sdk.go

```diff
@@ -0,0 +1,78 @@
+package sdk
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/inngest/inngest/inngest"
+	"github.com/inngest/inngest/pkg/function"
+)
+
+var (
+	ErrNoFunctions = fmt.Errorf("No functions registered within your app")
+)
+
+// RegisterRequest represents a new deploy request from SDK-based functions.
+// This lets us know that a new deploy has started and that we need to
+// upsert function definitions from the newly deployed endpoint.
+type RegisterRequest struct {
```

Comment:
**Thinking out loud** - this is the number one place where the comms could fail, where we mismatch expected data shapes.

Is there anything we could do programmatically or otherwise to ensure consistency across the various repos that'll use this?

Reply:
Hmm, a few things come to mind:

- We added the `v` tag specifically so that we can version this payload, allowing us to release new versions of this while maintaining backcompat.  This should minimize mismatched data once you've got a working implementation — this will only crop up when you're building a new SDK for the first time.
- We can & do report errors to the SDK via HTTP!